src/hooks/accounts/resolvers/query/activeAccount.tsx

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { Account } from '../../../../generated/graphql';
+import { useResolverToRef } from '../useAccountsMutationResolvers';
+import { GET_ACCOUNTS } from '../../queries/useGetAccountsQuery';
+import { usePersistActiveAccount } from '../../usePersistActiveAccount';
+import { find } from 'lodash';
+import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
+
+// make sure the __typename is well typed
+export const __typename: Account['__typename'] = 'Account';
+
+// helper function to decorate the extension entity for normalised caching
+const withTypename = (account: Account) => ({
+  __typename,
+  ...account,
+});
+
+export const useActiveAccountQueryResolver = () => {
+  const [persistedActiveAccount] = usePersistActiveAccount();
+
+  return {
+    activeAccount: useResolverToRef(
```

Comment:
i'd suggest looking at balances in terms of how the resolver is extracted as a factory, so it can be tested appropriately.

Apart from that, it does exactly what it should - checks for local storage active account (which is a contextual dependency) and then if there is an active account it fetches all accounts, good stuff.